src/Settings.qml

```diff
@@ -46,13 +46,18 @@ SilicaFlickable {
             width: parent.width
         }
 
-        TextSwitch {
-            text: "Enable compass"
-            checked: map.enable_compass
-            description: "Display a compass arrow under the GPS point" +
-            " (green points in the north direction)."
-            onClicked: map.enable_compass = checked
-            width: parent.width
+        ComboBox {
```

Comment:
May you add a label keeping the short description, like it's done for the GPS refresh rate slider for instance ?